.github/workflows/publish-api-helm-chart-workflow.yaml

```diff
@@ -0,0 +1,36 @@
+name: Publish OpenDORA API Helm Chart
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+    push:
+      tags:
+        - "*.*.*"
+      branches:
+        - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+
+      - name: Login to GitHub Container Registry
+        run: echo "${{ secrets.GHCR_TOKEN }}" | helm registry login ghcr.io -u ${{ github.repository_owner }} --password-stdin
```

Comment:
Attempted to login/publish with the GITHUB_TOKEN but this does not have required permissions for publishing, so a PAT needs to be provided instead.

Reply:
We have worked around this using [workflow permissions](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#defining-access-for-the-github_token-scopes):
```
permissions:
  contents: read
  id-token: read
  packages: write
```

We then used the [docker login action](https://github.com/docker/login-action?tab=readme-ov-file#github-container-registry).

That should also allow you to drop the `logout` step.

Hope this is helpful or interesting!

Reply:
Thanks for the tip, I'll give this a try. I think I assumed that `packages: write` was in the default permissions since we didn't have to specify any for publishing the NPM package to the GitHub registry.

Reply:
I re-read the docs for the docker login action which linked to [this page](https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio). I needed to set the package settings to allow write access from this repos actions!

Reply:
> id-token: read

Also `id-token: read` isn't a valid permission for actions apparently: https://github.com/github/docs/issues/26481